server/src/main/java/org/apache/iotdb/db/utils/QueryUtils.java

```diff
@@ -152,10 +142,11 @@ public static QueryDataSet getQueryDataSet(List<ShowTimeSeriesResult> timeseries
       updateRecord(record, result.getName());
       updateRecord(record, result.getAlias());
       updateRecord(record, result.getSgName());
-      updateRecord(record, result.getDataType());
-      updateRecord(record, result.getEncoding());
-      updateRecord(record, result.getCompressor());
-      updateRecord(record, result.getTagAndAttribute(), paths);
+      updateRecord(record, result.getDataType().toString());
+      updateRecord(record, result.getEncoding().toString());
+      updateRecord(record, result.getCompressor().toString());
```

Comment:
i think that too many type cast , how about used only in `serialize/deserialize`.